explorer/interpreter/interpreter.cpp

```diff
@@ -290,10 +290,9 @@ auto Interpreter::EvalPrim(Operator op, Nonnull<const Value*> /*static_type*/,
 auto Interpreter::CreateStruct(const std::vector<FieldInitializer>& fields,
                                const std::vector<Nonnull<const Value*>>& values)
     -> Nonnull<const Value*> {
-  CARBON_CHECK(fields.size() == values.size());
   std::vector<NamedValue> elements;
-  for (size_t i = 0; i < fields.size(); ++i) {
-    elements.push_back({fields[i].name(), values[i]});
+  for (const auto& [field, value] : llvm::zip_equal(fields, values)) {
```

Comment:
Throughout this patch:

Using `const auto&` here seems unnecessary to me: the iterator will be generating new values, so a non-reference would do the same thing. Maybe remove the `&`s throughout?

Reply:
Done